docs/guide/tutorials/gridfs.txt

```diff
@@ -1,3 +1,206 @@
+`md5 <https://en.wikipedia.org/wiki/MD5>`_ of the contents. There's also a
+``chunkSize`` field indicating that the file is
+broken up into chunks of size 255 kilobytes, which is the
+default.
+
+.. code-block:: js
+
+   > db.fs.chunks.count()
+   107
+
+Not surprisingly, 27847575/261120 is approximately 106.64, so the ``fs.chunks``
+collection contains 106 chunks with size 255KB and 1 chunk that's roughly
+255KB * 0.64. Each individual chunks document is similar to the document below.
+
+.. code-block:: js
+
+   > db.fs.chunks.findOne({}, { data: 0 })
+   {
+       "_id" : ObjectId("561fc381e81346c82d6397bc"),
+       "files_id" : ObjectId("561fc381e81346c82d6397bb"),
+       "n" : 0
+   }
+
+The chunk document keeps track of which file it belongs to and its order in
+the list of chunks. The chunk document also has a ``data`` field that contains
+the raw bytes of the file.
+
+You can configure both the chunk size and the ``fs`` prefix for the files and
+chunks collections at the bucket level. For instance, if you specify the
+``chunkSizeBytes`` and ``bucketName`` options as shown below, you'll get
+27195 chunks in the ``songs.chunks`` collection.
+
+.. code-block:: js
+
+   const bucket = new GridFSBucket(db, {
+     chunkSizeBytes: 1024,
+     bucketName: 'songs'
+   });
+
+
+   await new Promise((resolve, reject) => {
+     fs.createReadStream('./meistersinger.mp3')
+       .pipe(bucket.openUploadStream('meistersinger.mp3'))
+       .on('error', reject)
+       .on('finish', resolve);
+   });
+
+   console.log('done!');
+
+Downloading a File
+------------------
+
+Congratulations, you've successfully uploaded a file to MongoDB! However,
+a file sitting in MongoDB isn't particularly useful. In order to stream the
+file to your hard drive, an HTTP response, or to npm modules like
+`speaker <https://www.npmjs.com/package/speaker>`_\ , you're going to need
+a download stream. The easiest way to get a download stream is
+the ``openDownloadStreamByName()`` method.
+
+.. code-block:: js
+
+   const bucket = new GridFSBucket(db, {
+     chunkSizeBytes: 1024,
+     bucketName: 'songs'
+   });
+
+   await new Promise((resolve, reject) => {
+     bucket.openDownloadStreamByName('meistersinger.mp3')
+       .pipe(fs.createWriteStream('./output.mp3'))
+       .on('error', reject)
+       .on('finish', resolve);
+   });
+   
+   console.log('done!');
+
+Now, you have an ``output.mp3`` file that's a copy of the original
+``meistersinger.mp3`` file. The download stream also enables you to do some
+neat tricks. For instance, you can cut off the beginning of the song by
+specifying a number of bytes to skip. You can cut off the first 41 seconds of
+the mp3 and skip right to the good part of the song as shown below.
+
+.. code-block:: js
+
+   await new Promise((resolve, reject) => {
+     bucket.openDownloadStreamByName('meistersinger.mp3')
+       .start(1024 * 1585). // <-- skip the first 1585 KB, approximately 41 seconds
```

Comment:
there's a period at the end of .start and at the beginning of .pipe, so the full thing (ignoring the comment) reads
 ```
.start(1024 * 1585)..pipe(fs.createWriteStream('./output.mp3'))
```

the full code block should read:

```
await new Promise((resolve, reject) => {
  bucket.openDownloadStreamByName('meistersinger.mp3')
    .start(1024 * 1585) // <-- skip the first 1585 KB, approximately 41 seconds
    .pipe(fs.createWriteStream('./output.mp3'))
    .on('error', reject)
    .on('finish', resolve);
});

console.log('done!');
```